File: 大話/cpp/strategy/CashRebate.cpp
#include "CashRebate.h"

namespace strategy {

    CashRebate::CashRebate(double rebate) {
        this->rebate = rebate;
    }

    double CashRebate::getRebate() {
        return rebate;
    }

    void CashRebate::setRebate(double rebate) {
        this->rebate = rebate;
    }

    double CashRebate::acceptCash(double money) {
        return money * rebate / 10;
    }
}

File: 大話/sktp00023_bs/02策略模式/策略模式-商場管理軟體05簡單工廠和策略模式/商場管理軟體/CashRebate.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace 商場管理軟體
{
    //打折收費，繼承CashSuper
    class CashRebate : CashSuper
    {
        private double moneyRebate = 1d;
        //初始化時，必需要輸入折扣率,如八折，就是0.8
        public CashRebate(string moneyRebate)
        {
            this.moneyRebate = double.Parse(moneyRebate);
        }

        public override double acceptCash(double money)
        {
            return money * moneyRebate;
        }
    }
}
